server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1293,6 +1295,33 @@ public static Setting<Float> floatSetting(
         );
     }
 
+    private class FloatParser implements Function<String, Float>, Writeable {
+        private String key;
+        private float minValue;
+        private float maxValue;
+        private boolean isFiltered;
+
+        public FloatParser(String key, float minValue, float maxValue, boolean isFiltered) {
+            this.key = key;
+            this.minValue = minValue;
+            this.maxValue = maxValue;
+            this.isFiltered = isFiltered;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
```

Comment:
You need a constructor taking a `StreamInput` parameter and reading in the values corresponding to this output.  

Also write test methods exercising this class reading in and out.  (Same for all the other Parser implementations.)